File: 2010_2011_mf_sujet_5_monte_carlo_kolmogorov_smirnov/DKS.m
function [ ac ] = DKS(X,F0,alpha)
%UNTITLED Summary of this function goes here
%   Detailed explanation goes here
Y=F0(sort(X));
n=length(X);
Z1=[1:n]/n;
Z2=[0:n-1]/n;
T1=Z2-Y;
T2=Y-Z1;
d1=max(T1);
d2=max(T2);
d3=max(d1,d2)*sqrt(n);
ac=1-pks(d3);
if (ac>alpha) rep = 'OUI'
else rep = 'NON'
end
end

File: 2010_2011_mf_sujet_5_monte_carlo_kolmogorov_smirnov/DKS.py
import numpy as np
from scipy.stats import kstwobign


def DKS(X, F0, alpha):
    """One-sample Kolmogorov-Smirnov test. Returns p-value."""
    Y = F0(np.sort(X))
    n = len(X)
    Z1 = np.arange(1, n + 1) / n
    Z2 = np.arange(0, n) / n
    d1 = np.max(Z2 - Y)
    d2 = np.max(Y - Z1)
    d3 = max(d1, d2) * np.sqrt(n)
    ac = 1 - kstwobign.cdf(d3)
    rep = 'OUI' if ac > alpha else 'NON'
    print(rep)
    return ac
